web/source/osk/visualKeyboard.ts

```diff
@@ -679,6 +679,87 @@ namespace com.keyman.osk {
       }
     }
 
+    get width(): number {
+      if(this._width) {
+        return this._width;
+      } else {
+        return this.loadSizeFromCookie().width;
+      }
+    }
+
+    get height(): number {
+      if(this._height) {
+        return this._height;
+      } else {
+        return this.loadSizeFromCookie().height;
+      }
+    }
+
+    protected loadSizeFromCookie(): {width: number, height: number} {
+      let keyman = com.keyman.singleton;
+      let util = keyman.util;
+
+      var c = util.loadCookie('KeymanWeb_OnScreenKeyboard');
+      var newWidth: number, newHeight: number;
+      if(typeof(c) == 'undefined' || c == null) {
+        newWidth = screen.width * 0.3;
+        newHeight = 144; // 16 * 8; 16 is our preferred default font size.
+      } else {
+        // Restore OSK size - font size now fixed in relation to OSK height, unless overridden (in em) by keyboard
+        newWidth=util.toNumber(c['width'], 0.3 * screen.width); // Default - 30% of screen's width.
+
+        if(newWidth < 0.2*screen.width) {
+          newWidth = 0.2*screen.width;
+        } else if(newWidth > 0.9*screen.width) {
+          newWidth=0.9*screen.width;
+        }
+
+        newHeight=util.toNumber(c['height'],0.15*screen.height);
+
+        if(newHeight > 0.5*screen.height) {
+          newHeight=0.5*screen.height;
+        }
+      }
+
+      this.setSize(newWidth, newHeight);
```

Comment:
This is rather side-effecty and I'd prefer if we are cleaning up this function that we either refactor this, or that we make this function have no return value -- so that you'd first call `loadSizeFromCookie()` and then return look at `this.width`/`this.height`.

Reply:
I was originally going to do the latter, so I don't mind that approach.  I was afraid of this comment, just in the other direction.